pandas/_libs/tslibs/parsing.pyx

```diff
@@ -967,15 +970,18 @@ def guess_datetime_format(
         if set(attrs) & found_attrs:
             continue
 
-        if all(getattr(parsed_datetime, attr) is not None for attr in attrs):
-            for i, token_format in enumerate(format_guess):
-                token_filled = tokens[i].zfill(padding)
-                if (token_format is None and
-                        token_filled == parsed_datetime.strftime(attr_format)):
-                    format_guess[i] = attr_format
-                    tokens[i] = token_filled
-                    found_attrs.update(attrs)
-                    break
+        parsed_formatted = parsed_datetime.strftime(attr_format)
+        # The result of `.strftime("%Z")` on a dt with no timezone is ""
+        if len(parsed_formatted) == 0:
+            continue
+
+        for i, token_format in enumerate(format_guess):
+            token_filled = tokens[i].zfill(padding)
+            if token_format is None and token_filled == parsed_formatted:
+                format_guess[i] = attr_format
+                tokens[i] = token_filled
+                found_attrs.update(attrs)
```

Comment:
can you add a benchmark that supports this case (i think we have them for other things that guess the format)

Reply:
Sure!